jscripts/automention/automention.js

```diff
@@ -34,7 +48,15 @@ function automentionck( local ) {
 $(document).ready(function() {
 	if (typeof $.fn.sceditor !== 'undefined') {
 		if($('#message, #signature').sceditor("instance")) {
-			$($('#message, #signature').sceditor("instance").getBody()).atwho('setIframe').atwho(ment_settings);
+			var $iframe = $('.sceditor-container iframe');
+			$($('#message, #signature').sceditor("instance").getBody()).atwho('setIframe', $iframe[0], true).atwho(ment_settings);
+			var cssLink = $('<link/>',{
+				href: automention_css_file,
+				rel: 'stylesheet',
+				type: 'text/css'
+			});
+			var $iframeBody = $iframe.contents().find("body");
+			$iframeBody.prepend(cssLink);
```

Comment:
I didn't take the test. But this correction appears to be for sceditor only. Does this problem you are trying to solve only occur with sceditor?

Reply:
Yes, that's correct - it only occurred with SCEditor. More specifically, the problem was that when the editor was vertically scrolled, and the point at which the mention was being inserted lay below one page worth of scroll, the mention pop-up was vertically displaced below where it should have been by the amount of vertical scroll.

This was because `setIframe` was being called with no parameters. After adding the two parameters that you see - `$iframe[0]` and `/* asRoot = */true`, the vertical displacement of the pop-up was corrected, however, the pop-up then became unstyled - hence the code you see to inject its stylesheet into the iframe.

Reply:
thank you very much for the clarification and the correction.

Reply:
@Eldenroot and @martec, I identified the cause of one of the glitches that I mentioned in an earlier message in this PR's conversation tab. The root cause was that the `atwho` library appends its drop-down HTML into the `body` of the SCEditor iframe when the second argument to `setIframe` is true (as added above) - but that body should contain only actual post contents, which that added HTML corrupted. To solve this problem, I came up with an alternative fix for the original problem that the code above was supposed to solve (the misplacement of the drop-down), which keeps the drop-down HTML out of the post contents. It goes like this:

In `jquery.atwho.min.js`, replace:
```javascript
n.top+=i.top
```
With:
```javascript
n.top+=i.top-t(this.app.iframe).contents().find('html').scrollTop()
```
This is equivalent to replacing [line 821](https://github.com/ichord/At.js/blob/master/dist/js/jquery.atwho.js#L821) of the unminified file:
```javascript
      rect.top += iframeOffset.top - this.$inputor.scrollTop();
```
with:
```javascript
      rect.top += iframeOffset.top - $iframe.contents().find('html').scrollTop() - this.$inputor.scrollTop();
```

Then, in `automention.js`, within the `$(document).ready()` function, change the second newly-added argument to `setIframe` from `true` to `false`.

If you guys like, I will coordinate with @Eldenroot to update the PR.

Reply:
P.S. This makes the CSS-injection code unnecessary, so it can be removed from both `automention.js` and `automention.php` (the line which sets the `var automention_css_file` variable).

That code was anyway flawed in that it inserted the CSS into the iframe's `body` instead of its `head`. This is flawed both because (1) CSS links are technically valid only in `head` and not `body`, though the code seemed to work, and (2) again, it corrupts the contents of the post.